### ISSUE TITLE
Save prediction probabilities.

### ISSUE THREAD

**Issue by Koliham** [zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow#35]
Hi,
this code is great and so easy to implement, thanks!
I would like to have the probability for each class saved (with visual probability map, for each image, I would get 20 probability images (one for each class)). But just with getting a probability matrix would also be nice.
How do I have to modify the code to get this?

Thanks in advance!

**Comment by zhengyang-wang** [comment]
Try to save the results of this [line](https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow/blob/7439c8c6fa14cf946e0eef07a583cf5b98f572f5/model.py#L305). Note that the network outputs probability maps of 1/8 size and use bilinear interpolation to upsample them. So the upsampled maps may not be "probability maps" unless you normalize them again after upsampling.

**Comment by Koliham** [comment]
Thanks for the help! I got the "probabilities" with the line you described, but as you mentioned, no real probabilty values, because the values are in are range about 0-20.
Can you explain for me, how to normalize the values? Divide by a certain value?

**Comment by zhengyang-wang** [comment]
There is no perfect way. You can use softmax again to normalize the values. 
Or, you can try to remove the original softmax and put it after upsampling. This may require changing the loss function.
Actually, in the Deeplabv3 paper, it is recommended to calculate loss using upsampled logits and labels.
